pkg/telemetry/context.go

```diff
@@ -0,0 +1,50 @@
+package telemetry
+
+import (
+	"context"
+	"strings"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc"
+)
+
+type rpcContextName string
+
+const (
+	methodContextName  rpcContextName = "method"
+	serviceContextName rpcContextName = "service"
+)
+
+func saveMethodServiceInContext(ctx context.Context, method string, service string) context.Context {
+	methods := strings.Split(method, "/")
+
+	ctx = context.WithValue(ctx, methodContextName, methods[len(methods)-1])
+	return context.WithValue(ctx, serviceContextName, service)
+}
+
+// SaveRPCInfoInContext will save the rpc method and service information in context
+func SaveRPCInfoInContext(ctx context.Context) context.Context {
```

Comment:
It's a common idiom in Go to prefix context setters with `ContextWith...` . For example,

`ContextWithTimeout`
`ContextWithDeadline`
`ContextWithAuthClaims` (see https://github.com/openfga/openfga/blob/main/internal/authn/authn.go#L50-L52)

So, to follow status quo in Go world, can we refactor/rename as `ContextWithRPCInfo` convention.

Reply:
Similar, the "getter" idiom is usually of the form `<thing>FromContext`. For example,

`AuthClaimsFromContext`
`PrincipalFromContext`
etc..

So let's refactor/rename to `RPCInfoFromContext(context.Context)` as well for the getter.